src/server/tiered_storage.cc

```diff
@@ -355,6 +365,73 @@ void TieredStorage::Free(PrimeIterator it, DbTableStats* stats) {
   stats->tiered_size -= len;
 }
 
+size_t TieredStorage::GetBinSize(size_t blob_len) {
+  CHECK(blob_len <= kMaxSmallBin);
+  unsigned bin_index = SmallToBin(blob_len);
+  return kSmallBins[bin_index];
+}
+
+void TieredStorage::Defrag(DbIndex db_index) {
```

Comment:
I see this function is steel called from HeartBeat. you should not forloop untill !pages_to_defrag_.empty() because this can be making this very long step

Reply:
i remembered as we discussed last time, you mentioned this function will keep being preempted? so it is safe to let it run from beginning to the end?

Reply:
only if it runs on a dedicated fiber 
when I said this it was because we discussed calling it from a different fiber that will be invoked when we have free cpu as we have with defrug task, this is not the case in the current implementation